internal/infrastructure/kubernetes/proxy/resource_provider.go

```diff
@@ -120,7 +120,17 @@ func (r *ResourceRender) Service() (*corev1.Service, error) {
 	serviceSpec := resource.ExpectedServiceSpec(envoyServiceConfig)
 	serviceSpec.Ports = ports
 	serviceSpec.Selector = resource.GetSelector(labels).MatchLabels
-	serviceSpec.ExternalIPs = r.infra.Addresses
+
+	if (*envoyServiceConfig.Type) == egv1a1.ServiceTypeClusterIP {
+		if len(r.infra.Addresses) > 0 {
+			// we just need the first one address
+			// if address is not set, the automatically assigned clusterIP is used
+			serviceSpec.ClusterIP = r.infra.Addresses[0]
+			serviceSpec.ClusterIPs = r.infra.Addresses[0:1]
```

Comment:
should we assign all for `ClusterIPs` ?

Reply:
No, actually we can only set one clusterIPs, and clusterIPs[0] must be equal to clusterIP, and clusterIPs stipulates that each IP family can only set one IP.

**For example**: 
When we create the following Service resource and set multiple clusterIPs
```yaml
apiVersion: v1
kind: Service
metadata:
  name: envoy-custom
  namespace: envoy-gateway-system
spec:
  clusterIP: 10.102.168.100
  clusterIPs:
    - 10.102.168.100
    - 10.102.168.101
  ports:
  - name: envoy-http
    port: 10009
    protocol: TCP
    targetPort: 10000
  selector:
    app.kubernetes.io/component: proxy
    app.kubernetes.io/managed-by: envoy-gateway
    app.kubernetes.io/name: envoy
    gateway.envoyproxy.io/owning-gateway-name: eg-gw
    gateway.envoyproxy.io/owning-gateway-namespace: default
  type: ClusterIP
```

Actually api-server throws `The Service "envoy-custom" is invalid: spec.clusterIPs: Invalid value: []string{"10.102.168.100", "10.102.168.101"}: may specify no more than one IP for each IP family` error

So I think that when configuring `envoyService.type=ClusterIP`, if the user specifies addresses when configuring the gateway, we only need to get the first address.

I will add more detailed comments to the code explaining this

Reply:
thanks for explaining this, TIL about why `ClusterIPs` exist :)